File: 56-merge-intervals/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        l=[]
        intervals.sort()
        l.append(intervals[0])
        n = len(intervals)
        for i in range(1,n,1):
            if l[-1][1] >= intervals[i][0]:
                l[-1][1] = max( l[-1][1], intervals[i][1])
            else:
                l.append(intervals[i])
                
        return l
        

        

File: 56-merge-intervals/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>>ans;
         sort(intervals.begin(), intervals.end());
         int n = intervals.size();

         ans.push_back(intervals[0]);
         for(int i=1;i<n;i++)
         {

            if(ans.back()[1]>=intervals[i][0])
            {
                ans.back()[1] = max(ans.back()[1], intervals[i][1]);
            }
            else
            {
                ans.push_back(intervals[i]);
            }
         }
         return ans;

    }
};